File: ai-service-python/app/models/schema_definition.py
"""SchemaDefinition model - aligned with database schema."""

from sqlalchemy import String, Integer, BigInteger, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import BaseModel


class SchemaDefinition(BaseModel):
    __tablename__ = "schema_definition"

    tenant_id: Mapped[int] = mapped_column(BigInteger, nullable=False, index=True)
    name: Mapped[str] = mapped_column(String(128), nullable=False)
    description: Mapped[str] = mapped_column(String(512), nullable=True)
    dataset_id: Mapped[int] = mapped_column(BigInteger, nullable=True, index=True)
    columns: Mapped[str] = mapped_column(Text, nullable=True)  # JSON
    metrics: Mapped[str] = mapped_column(Text, nullable=True)  # JSON
    dimensions: Mapped[str] = mapped_column(Text, nullable=True)  # JSON
    config: Mapped[str] = mapped_column(Text, nullable=True)  # JSON
    version: Mapped[int] = mapped_column(Integer, default=1)
    status: Mapped[str] = mapped_column(String(16), default="active")

File: backend-java/src/main/java/com/enterprise/report/entity/SchemaDefinition.java
package com.enterprise.report.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("schema_definition")
public class SchemaDefinition {
    @TableId(type = IdType.ASSIGN_ID)
    private Long id;
    private Long tenantId;
    private String name;
    private String description;
    private Long datasetId;
    private String columns;
    private String metrics;
    private String dimensions;
    private String config;
    private Integer version;
    private String status;
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
    @TableLogic
    private Integer deleted;
}
